ddtrace/contrib/consul/patch.py

```diff
@@ -0,0 +1,52 @@
+import consul
+
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+from ...ext import AppTypes
+from ...pin import Pin
+from ...utils.wrappers import unwrap as _u
+
+
+_KV_FUNCS = ['Consul.KV.put', 'Consul.KV.get', 'Consul.KV.delete']
+
+
+def patch():
+    if getattr(consul, '__datadog_patch', False):
+        return
+    setattr(consul, '__datadog_patch', True)
+
+    pin = Pin(service='consul', app='consul', app_type=AppTypes.cache)
+    pin.onto(consul.Consul)
+    pin.onto(consul.Consul.KV)
```

Comment:
How are the multiple pins here intended to be used? Are both needed for the inheriting the pin used in `wrap_function`?

Reply:
The pin to `consul.Consul` could potentially be removed - this was done because [the consul lib implements the KV as a child class of Consul](https://github.com/cablehead/python-consul/blob/53eb41c4760b983aec878ef73e72c11e0af501bb/consul/base.py#L462) so I wasn't sure what correct practice is here.